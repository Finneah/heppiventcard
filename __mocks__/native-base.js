jest.mock('native-base', () => {
  return {
    StyleProvider: jest.fn().mockImplementation((x) => x),
  };
});
