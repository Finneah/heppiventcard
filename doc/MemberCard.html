<!DOCTYPE html>
<html lang="en">

<head>
  
  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <title>Heppiventcard MemberCard</title>

  <script src="https://cdn.jsdelivr.net/gh/google/code-prettify@master/loader/run_prettify.js"></script>
  <script src="https://unpkg.com/@babel/standalone/babel.min.js"></script>
  <script src="./build/entry.js"></script>
  <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.3.1/jquery.min.js"></script>
  <!--[if lt IE 9]>
    <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
  <![endif]-->
  <link href="https://fonts.googleapis.com/css?family=Roboto:100,400,700|Inconsolata,700" rel="stylesheet">
  <link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.6.3/css/all.css" integrity="sha384-UHRtZLI+pbxtHCWp1t77Bi1L4ZtiqrqD80Kn4Z8NTSRyMA2Fd33n5dQ8lWUE00s/" crossorigin="anonymous">
  <link type="text/css" rel="stylesheet" href="https://jmblog.github.io/color-themes-for-google-code-prettify/themes/tomorrow-night.min.css">
  <link type="text/css" rel="stylesheet" href="styles/app.min.css">
  <link type="text/css" rel="stylesheet" href="styles/iframe.css">
  <link type="text/css" rel="stylesheet" href="">
  <script async defer src="https://buttons.github.io/buttons.js"></script>

  
</head>



<body class="layout small-header">
    <div id="stickyNavbarOverlay"></div>
    

<div class="top-nav">
    <div class="inner">
        <a id="hamburger" role="button" class="navbar-burger" aria-label="menu" aria-expanded="false">
            <span aria-hidden="true"></span>
            <span aria-hidden="true"></span>
            <span aria-hidden="true"></span>
        </a>
        <div class="logo">
            
             
                <a href="index.html">
                    <h1 class="navbar-item">Heppiventcard</h1>
                </a>
            
        </div>
        <div class="menu">
            
            <div class="navigation">
                <a
                    href="index.html"
                    class="link"
                >
                    API Documentation
                </a>
                
                 
                    
                        <a
                            class="link user-link "
                            href="https://github.com/Finneah/heppiventcard"
                        >
                            Github
                        </a>
                    
                        <a
                            class="link user-link "
                            href="https://www.figma.com/file/Ze1jZrjApvyP6sv2e2Dpbb/heppiventcard?node-id=0%3A2932"
                        >
                            Example Application
                        </a>
                    
                
                
            </div>
        </div>
    </div>
</div>
    <div id="main">
        <div
            class="sidebar "
            id="sidebarNav"
        >
            
                <div class="search-wrapper">
                    <input id="search" type="text" placeholder="Search docs..." class="input">
                </div>
            
            <nav>
                
                    <h2><a href="index.html">Documentation</a></h2><div class="category"></div><div class="category"><h2>Component</h2><h3>Namespaces</h3><ul><li><a href="HomeScreenHeader.html">HomeScreenHeader</a></li><li><a href="MemberCard.html">MemberCard</a></li><li><a href="StampCard.html">StampCard</a></li></ul></div><div class="category"><h2>Database</h2><h3>Classes</h3><ul><li><a href="StampCardsModel.html">StampCardsModel</a></li><li><a href="StampCardsSchema.html">StampCardsSchema</a></li><li><a href="StampsModel.html">StampsModel</a></li><li><a href="StampsSchema.html">StampsSchema</a></li><li><a href="UserModel.html">UserModel</a></li><li><a href="UserSchema.html">UserSchema</a></li></ul></div><div class="category"><h2>View</h2><h3>Namespaces</h3><ul><li><a href="HomeScreen.html">HomeScreen</a></li></ul></div>
                
            </nav>
        </div>
        <div class="core" id="main-content-wrapper">
            <div class="content">
                <header class="page-title">
                    <p>Namespace</p>
                    <h1>MemberCard</h1>
                </header>
                




<section>

<header>
    
        <h2>MemberCard</h2>
        
    
</header>

<article>
    <div class="container-overview">
    
        
            <div class="description">MemberCard Element</div>
        

        


<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    
    

    

    

    
        <p class="tag-source">
            <a href="components_MemberCard.js.html" class="button">View Source</a>
            <span>
                <a href="components_MemberCard.js.html">components/MemberCard.js</a>, <a href="components_MemberCard.js.html#line11">line 11</a>
            </span>
        </p>
    
</dl>


        
    
    </div>
    
    

    

    

    

    

    

    

    
        <div class='vertical-section'>
            <h1>Methods</h1>
            <div class="members">
            
                <div class="member">


    
    <h4 class="name" id="._checkStampCount">
        <a class="href-link" href="#._checkStampCount">#</a>
        
            
                <span class='tag'>static</span>
            
        
        <span class="code-name">
            
                _checkStampCount<span class="signature">()</span><span class="type-signature"></span>
            
        </span>
    </h4>
    

    
    













<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    
    

    

    

    
        <p class="tag-source">
            <a href="components_MemberCard.js.html" class="button">View Source</a>
            <span>
                <a href="components_MemberCard.js.html">components/MemberCard.js</a>, <a href="components_MemberCard.js.html#line69">line 69</a>
            </span>
        </p>
    
</dl>





















</div>
            
                <div class="member">


    
    <h4 class="name" id="._getRank">
        <a class="href-link" href="#._getRank">#</a>
        
            
                <span class='tag'>static</span>
            
        
        <span class="code-name">
            
                _getRank<span class="signature">()</span><span class="type-signature"></span>
            
        </span>
    </h4>
    

    
    













<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    
    

    

    

    
        <p class="tag-source">
            <a href="components_MemberCard.js.html" class="button">View Source</a>
            <span>
                <a href="components_MemberCard.js.html">components/MemberCard.js</a>, <a href="components_MemberCard.js.html#line114">line 114</a>
            </span>
        </p>
    
</dl>





















</div>
            
                <div class="member">


    
    <h4 class="name" id="._renderContent">
        <a class="href-link" href="#._renderContent">#</a>
        
            
                <span class='tag'>static</span>
            
        
        <span class="code-name">
            
                _renderContent<span class="signature">()</span><span class="type-signature"></span>
            
        </span>
    </h4>
    

    
    













<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    
    

    

    

    
        <p class="tag-source">
            <a href="components_MemberCard.js.html" class="button">View Source</a>
            <span>
                <a href="components_MemberCard.js.html">components/MemberCard.js</a>, <a href="components_MemberCard.js.html#line141">line 141</a>
            </span>
        </p>
    
</dl>





















</div>
            
                <div class="member">


    
    <h4 class="name" id="._setUser">
        <a class="href-link" href="#._setUser">#</a>
        
            
                <span class='tag'>static</span>
            
        
        <span class="code-name">
            
                _setUser<span class="signature">()</span><span class="type-signature"></span>
            
        </span>
    </h4>
    

    
    













<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    
    

    

    

    
        <p class="tag-source">
            <a href="components_MemberCard.js.html" class="button">View Source</a>
            <span>
                <a href="components_MemberCard.js.html">components/MemberCard.js</a>, <a href="components_MemberCard.js.html#line101">line 101</a>
            </span>
        </p>
    
</dl>





















</div>
            
            </div>
        </div>
    

    

    
</article>

</section>




            </div>
            
            <footer class="footer">
                <div class="content has-text-centered">
                    <p>Documentation generated by <a href="https://github.com/jsdoc3/jsdoc">JSDoc 3.6.6</a></p>
                    <p class="sidebar-created-by">
                        <a href="https://github.com/SoftwareBrothers/better-docs" target="_blank">BetterDocs theme</a> provided with <i class="fas fa-heart"></i> by 
                        <a href="http://softwarebrothers.co" target="_blank">SoftwareBrothers - JavaScript Development Agency</a>
                    </p>
                </div>
            </footer>
            
        </div>
        <div id="side-nav" class="side-nav">
        </div>
    </div>
<script src="scripts/app.min.js"></script>
<script>PR.prettyPrint();</script>
<script src="scripts/linenumber.js"> </script>

<script src="scripts/search.js"> </script>

</body>
</html>