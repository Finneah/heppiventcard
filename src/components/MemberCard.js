import {Accordion, Body, Card, CardItem, Input, Item, Label} from 'native-base';
import React, {useState} from 'react';
import {StyleSheet} from 'react-native';
import GlobalColors from '../styles/GlobalColors';
import {strings} from '../i18n';
import {getData, storeData} from '../storage/AsyncStorage';
const MemberCard = ({}) => {
  const defaultMemberName = 'Your Name goes here';
  const [memberName, setMemberName] = useState();

  // _setName();
  React.useLayoutEffect(() => {
    _setName();
  }, []);

  async function _setName() {
    var name = await getData('NAME');

    setMemberName(name);
  }
  function _renderContent() {
    return (
      <CardItem style={styles.cardItemStyle}>
        <Body>
          <Item
            // eslint-disable-next-line react-native/no-inline-styles
            style={{
              borderColor:
                memberName === '' ? GlobalColors.brandPrimary : 'transparent',
            }}
            inlineLabel>
            <Label style={{color: GlobalColors.brandPrimary}}>
              {strings('NAME')}
            </Label>
            <Input
              style={styles.nameInput}
              placeholder={defaultMemberName}
              placeholderTextColor={GlobalColors.brown}
              value={memberName}
              onChangeText={(text) => {
                storeData('NAME', text);
                setMemberName(text);
              }}
            />
          </Item>
          <Item style={styles.transparentBorder} inlineLabel>
            <Label style={{color: GlobalColors.brandPrimary}}>
              {strings('RANK')}
            </Label>
            <Input
              disabled
              style={styles.rankInput}
              placeholder={'Gast'}
              value={'Gast'}
            />
          </Item>
        </Body>
      </CardItem>
    );
  }

  return (
    <Card>
      <CardItem first last>
        <Accordion
          style={styles.transparentBorder}
          dataArray={[{title: strings('MEMBERCARD')}]}
          animation={true}
          expanded={0}
          headerStyle={[styles.transparentBG]}
          icon="chevron-forward"
          expandedIcon="chevron-down"
          iconStyle={{color: GlobalColors.brandPrimary}}
          expandedIconStyle={{color: GlobalColors.brandSecondary}}
          renderContent={_renderContent}
        />
      </CardItem>
    </Card>
  );
};
const styles = StyleSheet.create({
  cardItemStyle: {
    alignContent: 'center',
    justifyContent: 'center',
    // backgroundColor: GlobalColors.brown,
  },
  nameInput: {
    fontSize: 20,
    fontWeight: 'bold',
    color: GlobalColors.brandPrimary,
  },
  rankInput: {
    fontSize: 20,
    fontWeight: 'bold',
    color: GlobalColors.brandSecondary,
  },
  transparentBorder: {
    borderColor: 'transparent',
  },
});
export default MemberCard;
