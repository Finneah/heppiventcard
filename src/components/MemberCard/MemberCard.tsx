import {Stack, Heading, HStack} from 'native-base';
import * as React from 'react';

import {strings} from '../../locale/i18n';

import {Stamps, User} from '../../database';
import {StampsModel} from '../../database/Models/StampsModel';
import {UserType} from '../../Helper/Types';
import {isValidUser, _getRank} from '../../Helper/Helper';
import {UserModel} from '../../database/Models/UserModel';

let stampsModel = new StampsModel();
let userModel = new UserModel();
/**
 * @category Component
 * @namespace MemberCard
 * @description MemberCard Element
 */
const MemberCard: React.FC = () => {
  let baseUser: UserType = {name: strings('NAME_PLACEHOLDER'), rank: 0};
  const [user, setUser] = React.useState(baseUser);

  React.useLayoutEffect(() => {
    /**
     * @memberof MemberCard
     */
    User.onLoaded(async () => {
      try {
        if (User.data().length === 0) {
          var newUser: UserType = {name: strings('NAME_PLACEHOLDER'), rank: 0};
          userModel.createNewUser(newUser);
          setUser(newUser);
        } else {
          if (isValidUser(User.data()[0])) {
            setUser(User.data()[0]);
          }
        }
      } catch (error) {
        console.info(error);
      }
    });

    /**
     * @memberof MemberCard
     */
    User.onChange(() => {
      try {
        if (isValidUser(User.data()[0])) {
          setUser(User.data()[0]);
        }
      } catch (error) {
        console.info(error);
      }
    });

    /**
     * @memberof MemberCard
     */
    Stamps.onChange(() => {
      try {
        var stampsLength = stampsModel._getStampsLength();
        if (stampsLength) {
          let dbUser: UserType = User.data()[0];
          dbUser.rank = stampsLength;
          userModel.updateUser(dbUser);
        }
      } catch (error) {
        console.warn('Member Card Stamps.onChange', error);
      }
    });
  }, []);

  /**
   * @memberof MemberCard
   */

  // function _renderContent() {
  //   return (
  //     <Box border={1} borderRadius="md">
  //       <Stack direction="column" space={4} divider={<Divider />}>
  //         <Heading>{user ? user.name : strings('NAME_PLACEHOLDER')}</Heading>
  //       </Stack>
  //     </Box>
  //     // <CardItem>
  //     //   <Body>
  //     //     <Item
  //     //       // eslint-disable-next-line react-native/no-inline-styles
  //     //       style={{
  //     //         borderColor:
  //     //           user && user.name === ''
  //     //             ? GlobalColors.brandPrimary
  //     //             : 'transparent',
  //     //       }}
  //     //       inlineLabel>
  //     //       <Label style={{color: GlobalColors.brandPrimary}}>
  //     //         {strings('NAME')}
  //     //       </Label>

  //     //       <Input
  //     //         style={styles.nameInput}
  //     //         placeholder={defaultMemberName}
  //     //         placeholderTextColor={GlobalColors.brown}
  //     //         value={user ? user.name : ''}
  //     //         onChangeText={(text) => {
  //     //           /**
  //     //            * @todo update User
  //     //            *
  //     //            */

  //     //           User.update(user.id, {...user, name: text}, true);
  //     //           setUser({...user, name: text});
  //     //         }}
  //     //       />
  //     //     </Item>

  //     //     <Item style={[styles.transparentBorder]} inlineLabel>
  //     //       <Label style={{color: GlobalColors.brandPrimary}}>
  //     //         {strings('RANK')}
  //     //       </Label>
  //     //       <Input
  //     //         disabled
  //     //         style={styles.rankInput}
  //     //         placeholder={strings('RANK_00')}
  //     //         value={_getRank()}
  //     //       />
  //     //     </Item>
  //     //   </Body>
  //     // </CardItem>
  //   );
  // }

  return (
    <Stack
      direction="row"
      px={5}
      justifyContent="space-between"
      alignItems="center">
      <HStack alignItems="center">
        <Heading size="sm">{strings('HELLO') + ' '}</Heading>
        <Heading size="sm">{user?.name ?? ''}</Heading>
      </HStack>

      <Heading size="sm" color="secondary.400">
        {_getRank(user)}
      </Heading>
    </Stack>
  );
};

export default MemberCard;
export const util: any = {forEach: null};
