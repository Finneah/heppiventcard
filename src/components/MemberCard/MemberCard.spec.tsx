import 'react-native';
import * as React from 'react';
import {render} from '@testing-library/react-native';

import MemberCard from './MemberCard';
import {NativeBaseProvider} from 'native-base';

const wrapper = ({children}: any) => (
  <NativeBaseProvider
    initialWindowMetrics={{
      frame: {x: 0, y: 0, width: 0, height: 0},
      insets: {top: 0, left: 0, right: 0, bottom: 0},
    }}>
    {children}
  </NativeBaseProvider>
);

describe('MemberCard', () => {
  test('renders correctly', () => {
    const memberCard = render(<MemberCard />, {wrapper});
    expect(memberCard.toJSON()).toMatchSnapshot();
  });

  test('userNameHeading is NAME_PLACEHOLDER ', () => {
    const {getByText} = render(<MemberCard />, {wrapper});
    const userNameHeading = getByText('NAME_PLACEHOLDER');
    expect(userNameHeading).toBeTruthy();
  });
  test("helloHeading is HELLO + ' '", () => {
    const {getByText} = render(<MemberCard />, {wrapper});
    const helloHeading = getByText('HELLO' + ' ');
    expect(helloHeading).toBeTruthy();
  });

  // test('validate User', () => {
  //   var validUser = isValidUser({name: 'XY', rank: 0});
  //   expect(validUser).toBeTruthy();

  //   // const mockCallback = jest.fn((x) => 42 + x);
  //   // util.forEach([0, 1], mockCallback);
  // });
});
