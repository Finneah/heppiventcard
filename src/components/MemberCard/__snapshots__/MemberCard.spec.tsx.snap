// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MemberCard renders correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    alignItems="center"
    flexDirection="row"
    justifyContent="space-between"
    px={5}
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingLeft": 20,
          "paddingRight": 20,
        },
      ]
    }
  >
    <View
      alignItems="center"
      flexDirection="row"
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
          },
        ]
      }
    >
      <Text
        color="muted.700"
        fontFamily="body"
        fontSize="md"
        fontStyle="normal"
        fontWeight="bold"
        style={
          Array [
            Object {
              "color": "#404040",
              "fontSize": 16,
              "fontStyle": "normal",
              "fontWeight": "700",
            },
          ]
        }
      >
        HELLO 
      </Text>
      <View
        style={
          Array [
            Object {},
          ]
        }
      />
      <Text
        color="muted.700"
        fontFamily="body"
        fontSize="md"
        fontStyle="normal"
        fontWeight="bold"
        style={
          Array [
            Object {
              "color": "#404040",
              "fontSize": 16,
              "fontStyle": "normal",
              "fontWeight": "700",
            },
          ]
        }
      >
        NAME_PLACEHOLDER
      </Text>
    </View>
    <View
      style={
        Array [
          Object {},
        ]
      }
    />
    <Text
      color="secondary.400"
      fontFamily="body"
      fontSize="md"
      fontStyle="normal"
      fontWeight="bold"
      style={
        Array [
          Object {
            "color": "#831843",
            "fontSize": 16,
            "fontStyle": "normal",
            "fontWeight": "700",
          },
        ]
      }
    >
      RANK_00
    </Text>
  </View>
</RNCSafeAreaProvider>
`;
